File: Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double num1, num2, result;
        char operator;

        System.out.println("Simple Calculator");

        while (true) {
            System.out.print("Enter first number: ");
            num1 = scanner.nextDouble();

            System.out.print("Enter operator (+, -, *, /): ");
            operator = scanner.next().charAt(0);

            System.out.print("Enter second number: ");
            num2 = scanner.nextDouble();

            switch (operator) {
                case '+':
                    result = num1 + num2;
                    System.out.println(num1 + " + " + num2 + " = " + result);
                    break;
                case '-':
                    result = num1 - num2;
                    System.out.println(num1 + " - " + num2 + " = " + result);
                    break;
                case '*':
                    result = num1 * num2;
                    System.out.println(num1 + " * " + num2 + " = " + result);
                    break;
                case '/':
                    if (num2 == 0) {
                        System.out.println("Error: Division by zero is not allowed.");
                        continue;
                    }
                    result = num1 / num2;
                    System.out.println(num1 + " / " + num2 + " = " + result);
                    break;
                default:
                    System.out.println("Error: Invalid operator.");
                    continue;
            }

            System.out.print("Do you want to continue? (yes/no): ");
            if (!scanner.next().equalsIgnoreCase("yes")) {
                break;
            }
        }

        scanner.close();
    }
}

File: calculator.rb
class Calculator
  def add(a, b)
    a + b
  end

  def subtract(a, b)
    a - b
  end

  def multiply(a, b)
    a * b
  end

  def divide(a, b)
    return "Error: Division by zero" if b == 0
    a.to_f / b
  end

  def power(a, b)
    a ** b
  end

  # Performs a calculation based on the specified operation
  #
  # @param operation [String] The operation to perform (add/+, subtract/-, multiply/*, divide//, power/**)
  # @param a [Numeric] The first operand
  # @param b [Numeric] The second operand
  # @return [Numeric, String] The calculation result or an error message
  def calculate(operation, a, b)
    case operation.downcase
    when 'add', '+'
      add(a, b)
    when 'subtract', '-'
      subtract(a, b)
    when 'multiply', '*'
      multiply(a, b)
    when 'divide', '/'
      divide(a, b)
    when 'power', '**'
      power(a, b)
    else
      "Error: Unknown operation"
    end
  end
end

# Interactive calculator
def run_calculator
  calc = Calculator.new
  
  puts "Ruby Calculator"
  puts "Operations: add, subtract, multiply, divide, power"
  puts "Type 'quit' to exit"
  
  loop do
    print "\nEnter operation: "
    operation = gets.chomp
    break if operation.downcase == 'quit'
    
    print "Enter first number: "
    a = gets.chomp.to_f
    
    print "Enter second number: "
    b = gets.chomp.to_f
    
    result = calc.calculate(operation, a, b)
    puts "Result: #{result}"
  end
end

# Run the calculator if this file is executed directly
if __FILE__ == $0
  run_calculator
end
